src/staticanalysis/bot/static_analysis_bot/workflows/remote.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from cli_common.log import get_logger
+from static_analysis_bot.config import settings
+from static_analysis_bot.lint import MozLintIssue
+
+logger = get_logger(__name__)
+
+
+ISSUE_MARKER = 'TEST-UNEXPECTED-ERROR | '
+
+
+class AnalysisTask(object):
+    '''
+    An analysis CI task running on Taskcluster
+    '''
+    def __init__(self, task_id):
+        self.task_id = task_id
+        self.run_id = None
+        self.task = None
+
+    def load_logs(self, queue_service):
+
+        # Load base task
+        self.task = queue_service.task(self.task_id)
+        self.name = self.task['metadata'].get('name', 'unknown')
+        logger.info('Lookup task dependency', id=self.task_id, name=self.name)
+
+        # Load task status
+        status = queue_service.status(self.task_id)
+        assert 'status' in status, 'No status data for {}'.format(self.task_id)
+        state = status['status']['state']
+
+        # Process only the failed tasks
+        # A completed task here means the analyzer did not find any issues
+        if state == 'completed':
+            logger.info('No issues detected by completed task', id=self.task_id)
+            return
+        elif state != 'failed':
+            logger.warn('Unsupported task state', state=state, id=self.task_id)
+            return
+
+        # Load artifact logs from the last run
+        self.run_id = status['status']['runs'][-1]['runId']
+        artifacts = queue_service.listArtifacts(self.task_id, self.run_id)
+        assert 'artifacts' in artifacts, 'Missing artifacts'
+        logs = [
+            artifact['name']
+            for artifact in artifacts['artifacts']
+            if artifact['storageType'] != 'reference' and artifact['contentType'].startswith('text/')
+        ]
+
+        # Load logs from artifact API
+        out = {}
+        for log in logs:
+            logger.info('Reading log', task_id=self.task_id, log=log)
+            try:
+                artifact = queue_service.getArtifact(self.task_id, self.run_id, log)
```

Comment:
Does `getArtifact` retry? Given that they are stored in S3, we should retry a few times in case of failure.

Reply:
The taskcluster python client [supports retries](https://github.com/taskcluster/taskcluster-client.py/blob/master/taskcluster/client.py#L460), with a default of [5 retries](https://github.com/taskcluster/taskcluster-client.py/blob/8c040e04f72dd0214c394c2ba25d342f10937226/taskcluster/client.py#L36)